lib/Doctrine/DBAL/Driver/LastInsertId.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Doctrine\DBAL\Driver;
+
+/**
+ * Last insert ID container.
+ */
+final class LastInsertId
+{
+    /** @var string */
+    private $value = '0';
+
+    public function get() : string
+    {
+        return $this->value;
+    }
+
+    public function register(string $value) : void
```

Comment:
I'm not a fan of having this crate mutable.

Reply:
We can make it immutable. Would it make more sense to rename this to something like `with() : self` in this case? What about `get()`? Should it be something like `toValue()` instead?

Reply:
@Majkl578, sadly, the immutable approach will not just work since `MysqliConnection` and `MysqliStatement` share the same instance of `LastInsertId`. It's probably done like this to mitigate:

> MySQL drivers reset the last insert ID to `0` after each non-insert query

It could be worked around by keeping the reference to the connection from the statement. But this approach stinks too (also implemented in PDO Statement/Connection pair).